Source/ProgressionSystemRuntime/Private/Widgets/PSMenuWidget.cpp

```diff
@@ -0,0 +1,76 @@
+﻿// Copyright (c) Valerii Rotermel & Yevhenii Selivanov
+
+#include "Widgets/PSMenuWidget.h"
+//---
+
+//---
+#include "Data/PSDataAsset.h"
+#include "Blueprint/WidgetTree.h"
+#include "Components/HorizontalBox.h"
+#include "Components/HorizontalBoxSlot.h"
+#include "Components/Image.h"
+#include "Data/PSWorldSubsystem.h"
+
+#include UE_INLINE_GENERATED_CPP_BY_NAME(PSMenuWidget)
+
+void UPSMenuWidget::AddImagesToHorizontalBox(int32 AmountOfUnlockedPoints, int32 AmountOfLockedPoints)
+{
+	checkf(HorizontalBox, TEXT("ERROR: 'HorizontalBox' is null"));
+	if (HorizontalBox)
+	{
+		// Loop to create and add images to the Horizontal Box for unlocked stars
+		for (int32 i = 0; i < AmountOfUnlockedPoints; i++)
+		{
+			UImage* ImageWidget = WidgetTree->ConstructWidget<UImage>(UImage::StaticClass());
+			UPSDataAsset::Get(); // performs checkif on get, so if null it will crash 
```

Comment:
Please remove line #25. You say you want to perform `checkf` there, but you are already calling `UPSDataAsset::Get()` on the next line that will perform own `checkf` as well.

Reply:
done